src/lib.rs

```diff
@@ -596,27 +596,40 @@ where
 {
     let samples: Vec<_> = items
         .into_iter()
+        .take(t)
```

Comment:
Good idea, but unfortunately `Take<I>` is only `TrustedLen` if `I` also is, so in general we can't rely on that.

Reply:
Right! My omission. It's actually even worse. It requires an nightly feature `trusted_len`.

I'm not sure if `<` is enough for error checking. What if the number of shares is greater than `t`? That should be an error, although a bizzarre one. :thinking:

Reply:
I actually think it's fine to not make that an error, although maybe we should check that the redundant entries do match the same resulting polynomial?

Anyway, as a user I'd expect it to work if e.g. everyone (and not just exactly `t `) provided a signature/decryption share.

Reply:
Then it's fine as it is.

Reply:
A problem with such a check is that each redundant value's verification would involve as many arithmetic operations in the curve (i.e. the slow ones) as the interpolation itself, so it would be quite an overhead. And if the user has verified the shares beforehand, that overhead isn't even necessary.